onnxruntime/core/session/inference_session.h

```diff
@@ -263,7 +263,7 @@ class InferenceSession {
   common::Status Run(const RunOptions& run_options, const std::vector<std::string>& feed_names,
                      const std::vector<OrtValue>& feeds, const std::vector<std::string>& output_names,
                      std::vector<OrtValue>* p_fetches,
-                     const std::vector<OrtDevice>* p_fetches_device_info = nullptr) ORT_MUST_USE_RESULT;
+                     const std::vector<OrtDevice>* p_fetches_device_info = nullptr, int64_t run_id = DEFAULT_RUN_ID) ORT_MUST_USE_RESULT;
```

Comment:
I'd prefer to keep `run_id` out of the public API, since it exposes partial run by the backdoor. Instead of adding a default-valued argument here, add a private method taking `run_id` that both `Run()` and `PartialRun()` call.

Reply:
This is not a public API. `run_id` is not exposed as part of any public API.

---
In reply to: [600935606](https://github.com/microsoft/onnxruntime/pull/7077#discussion_r600935606) [](ancestors = 600935606)

Reply:
It's a public method on `InferenceSession` - even if this class isn't covered by a formal backwards compatibility guarantee, it is one of our core classes, and we should apply high standards to changing its public methods.

Reply:
Public method is not the same as public API (as your initial comment referred). Yes, this is a core class but the run method has been overloaded several times during the course of time. Inference team will review this PR and I will seek their opinion on this to do the right thing. Thanks!